python/modules/variable.py

```diff
@@ -35,25 +36,46 @@ def get_variable(self, env):
     def get_value(self, env, args = {}):
         return env[self.value].get_value(env, args)
 
+class VariablePath(Variable):
+    def __init__(self, key, value, args = {}):
+        Variable.__init__(self, key, value, args = {})
+        self.value = FM.path_unify(value)
+
 class VariableParted(Variable):
     def __init__(self, key, value, args = {}):
         Variable.__init__(self, key, value, args = {})
 
+    def check_paths_presence(self, var_dict):
+        for key, variable in var_dict.items():
+            if isinstance(variable, VariablePath):
+                return True
+        return False
+
+    def get_value(self, env, args = {}):
+        return self._get_value(env, args = {})[0]
+
     def get_variable(self, env):
         # creates variable with merged value
-        merged_value = self.get_value(env)
-        var = Variable(self.key, merged_value)
+        merged_value, contains_path = self._get_value(env)
+        if contains_path:
+            var = VariablePath(self.key, merged_value)
+        else:
+            var = Variable(self.key, merged_value)
         return var
 
-    def get_value(self, env, args = {}):
+    def _get_value(self, env, args = {}):
```

Comment:
class VariablePath have global function get_value and local function _get_value().
It's misleading. Please, add comments to all functions of class Variable, because I do not know their functionality. I don't know how to use them in xml